--- ../avorion-stock/mods/DccMothballing/Commands/Disable.lua	1969-12-31 18:00:00.000000000 -0600
+++ ../avorion-mothballing/mods/DccMothballing/Commands/Disable.lua	2018-02-17 13:15:17.468000000 -0600
@@ -0,0 +1,46 @@
+--[[----------------------------------------------------------------------------
+AVORION: Mothballing
+darkconsole <darkcee.legit@gmail.com>
+
+This script disables the script that puts a ship in mothballs.
+----------------------------------------------------------------------------]]--
+
+function initialize()
+
+	if(onServer())
+	then
+		local Ship = Entity(Player().craftIndex)
+		local Script = "mods/DccMothballing/Entity/MothballedShip.lua"
+
+		if(Ship == nil or Ship.type ~= EntityType.Ship)
+		then
+			print("[Mothballs] You are not currently controlling a ship.")
+			return terminate()
+		end
+
+		Ship:removeScript(Script)
+		print("[Mothballs] " .. Ship.name .. " is no longer mothballed.")
+
+		-- ping the client to show a message about it.
+		deferredCallback(
+			1, "WeDoneHereAndThere",
+			"Back From Retirement",
+			"The " .. Ship.name .. " now needs a full crew."
+		)
+
+		return
+	end
+end
+
+function WeDoneHereAndThere(Title,Text)
+	if(onServer())
+	then
+		invokeClientFunction(Player(),"WeDoneHereAndThere",Title,Text)
+		terminate()
+		return
+	end
+
+	displayMissionAccomplishedText(Title,Text)
+	terminate()
+	return
+end
